.github/workflows/CI.yml

```diff
@@ -86,92 +84,61 @@ jobs:
       - name: Get current date and time
         id: get_datetime
         run: |
-          echo "::set-output name=DATE::$(date +'%Y%m%d')"
-          echo "::set-output name=TIME::$(date +'%H%M')"
           echo "::set-output name=DATETIME::$(date +'%Y%m%d')$(date +'%H%M')"
 
   ############################################################################
   # Unit tests                                                               #
   ############################################################################
   unit-tests:
     name: Unit Tests
-    needs: prepare_ci_run
     runs-on: ubuntu-20.04
     steps:
-      - name: Set up Go 1.x
-        uses: actions/setup-go@v2
-        with:
-          go-version: ^1.13
-      - name: Checkout Code
-        uses: actions/checkout@v2.3.4
-
-      - name: Load CI Environment from .ci_env
-        id: load_ci_env
-        uses: c-py/action-dotenv-to-setenv@v3
-        with:
-          env-file: .ci_env
-
-      - name: Test
-        run: go test -coverprofile=coverage.txt -covermode=atomic -v ./...
-        working-directory: .
-
+      - name: Check out repository
+        uses: actions/checkout@v2
+      - name: Unit tests
+        uses: ./.github/actions/unit-tests
 
   ############################################################################
   # Build Docker Image                                                      #
   ############################################################################
   docker_build:
-    needs: [prepare_ci_run, unit-tests]
+    needs: [prepare_ci_run]
```

Comment:
Do you really want to start building the docker image before the unit tests are done and have passed?

Reply:
Yes. It speeds up the process by a lot, and to be honest, for integrations I'd rather have the Docker Build quickly to try it out, if needed.

Reply:
ok, makes sense 👍🏼